quintilesims/layer0
# WIP - Adds rudimentary support for Application Load Balancers

**What does this pull request do?**
Adds a new type of Load Balancer: Application Load Balancer. Default behavior has been kept the same but you can now optionally specify `--type elb|alb` flag (default is elb).

An application load balancer has health check that needs to be specified differently to classic ELB's health check target. So a new property `Path` has been added to HealthCheck model which is used for an ALB type Load Balancer.

When creating a fargate service, you will need to specify a alb load balancer as a classic ELB isn't supported.

**How should this be tested?**
* Create a new Applicaiton Load Balancer using `--type alb` flag
* Specify the load balancer when creating a Fargate service
* Ensure that the service is accessible and Target Group health check recognizes a healthy service

**Todo**

- [x] Read SSL certificate information from Target Group Targets (how?)
- [x] Add wait logic when creating/deleting load balancers
- [ ] Update Load Balancer and Service unit tests for ALB cases
- [x] Update Service Read to check if the service is behind an ALB and resolve LB correctly
